includes/Data/ContentBlocksResolver.php

```diff
@@ -242,6 +243,33 @@ private static function populate_post_content_inner_blocks( array $block ): arra
 		return $block;
 	}
 
+	/**
+	 * Populates the innerBlocks of this block with navigation item blocks from the referenced navigation post.
+	 *
+	 * @param array<string,mixed> $block The block to populate.
+	 *
+	 * @return array<string,mixed> The populated block.
+	 */
+	private static function populate_navigation_blocks( array $block ): array {
+		if ( 'core/navigation' !== $block['blockName'] || ! isset( $block['attrs']['ref'] ) ) {
+			return $block;
+		}
+
+		$navigation_post = get_post( $block['attrs']['ref'] );
```

Comment:
Should we check vs type or just typecast the `$block['attrs']['ref']` ?
int|WP_Post

Reply:
I don't think this is necessary since this is how Gutenberg checks it as well

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/index.php#L230-L241

Reply:
I would cast it to `(int)` just so nobody needs to monitor if wp decides to change the `ref` to a numeric string.